azure-iot-device/azure/iot/device/iothub/sync_clients.py

```diff
@@ -369,23 +369,45 @@ def receive_twin_desired_properties_patch(self, block=True, timeout=None):
         logger.info("twin patch received")
         return patch
 
-    # @property
-    # def on_method_request_received(self):
-    #     return self._on_method_request_received
+    def _generic_handler_setter(self, handler_name, feature_name, new_handler):
+        self._validate_receive_handler_setter()
```

Comment:
>_validate_receive_handler_setter [](start = 13, length = 32)

slight name confusion: _validate_receive_handler_setter in this context makes it sound like you're validating the setter itself.  maybe _validate_use_of_receive_handlers and _validate_use_of_receive_apis?